js/zb.js

```diff
@@ -1179,14 +1179,25 @@ module.exports = class zb extends Exchange {
 
     parseOHLCV (ohlcv, market = undefined) {
         if (market['swap']) {
-            return [
-                this.safeInteger (ohlcv, 5),
-                this.safeNumber (ohlcv, 0),
-                this.safeNumber (ohlcv, 1),
-                this.safeNumber (ohlcv, 2),
-                this.safeNumber (ohlcv, 3),
-                this.safeNumber (ohlcv, 4),
-            ];
+            if (this.safeInteger (ohlcv, 5)) {
+                return [
+                    this.safeInteger (ohlcv, 5),
```

Comment:
Let's not do the same safeInteger check twice, instead, only do it once.
Also, we should not do `if (safeInteger (ohlcv, 5))`, instead we always compare to `undefined`: `if (safeInteger (ohlcv, 5) !== undefined)`, otherwise a proper integer value of 0 will be coerced to boolean `false`.

```JavaScript
            const candleLength = ohlcv.length;
            if (candleLength > 5) {
                // a regular candle
                return [
                    this.safeInteger (ohlcv, 5),
                    ...
            } else {
                // index/mark candle
                ...
```